pkg/dataobj/planner/physical/expressions.go

```diff
@@ -0,0 +1,181 @@
+package physical
+
+import "fmt"
+
+type ExpressionType uint32
+
+const (
+	ExprTypeUnary ExpressionType = iota
+	ExprTypeBinary
+	ExprTypeLiteral
+	ExprTypeColumn
+)
+
+func (t ExpressionType) String() string {
+	switch t {
+	case ExprTypeUnary:
+		return "UnaryExpression"
+	case ExprTypeBinary:
+		return "BinaryExpression"
+	case ExprTypeLiteral:
+		return "LiteralExpression"
+	case ExprTypeColumn:
+		return "ColumnExpression"
+	default:
+		return fmt.Sprintf("unknown expression type %d", t)
+	}
+}
+
+type UnaryOpType uint32
+
+const (
+	UnaryOpNot UnaryOpType = iota
+	UnaryOpAbs
+)
+
+func (t UnaryOpType) String() string {
+	switch t {
+	case UnaryOpNot:
+		return "NOT"
+	case UnaryOpAbs:
+		return "ABS"
+	default:
+		return fmt.Sprintf("unknown unary operator type %d", t)
+	}
+}
+
+type BinaryOpType uint32
+
+const (
+	BinaryOpEq BinaryOpType = iota
+	BinaryOpNeq
+	BinaryOpGt
+	BinaryOpGte
+	BinaryOpLt
+	BinaryOpLte
+	BinaryOpAnd
+	BinaryOpOr
+	BinaryOpXor
+	BinaryOpNot
+	BinaryOpAdd
+	BinaryOpSub
+	BinaryOpMul
+	BinaryOpDiv
+	BinaryOpMod
+	BinaryOpMatchStr
+	BinaryOpNmatchStr
```

Comment:
nit: when I read this, `Nmatch` blends in with `Match` and almost looks like a typo :) `NoMatch`? `NotMatch`?